### DISCUSSION TITLE
Given a file id, is it possible to find the system path on the host system?

### DISCUSSION THREAD

**Discussion by tbrittoborges** [cloud-py-api/nc_py_api#244] [Q&A] [answered]
I would like to find the path to a given file on the host system:

```{python}
from nc_py_api import NextcloudApp

nc = NextcloudApp(
    base_url="https://your-nextcloud-instance.com",
    user="your-username",
    password="your-password"
)

# 
file_id = 123456
file_path = nc.files.get_path(file_id)
```
I understand there are permissions issues, so assuming the users has admin privileges, would it be possible to find the host system path to a file stored? 

We have a solution using the Nextcloud database, but we would prefer to replace that.

**Comment by bigcat88** [comment]
Do you need to find a file by `fileid` and you don't know to which Nextcloud user it belongs?

If this is the question, then the reply is no, using Nextcloud WebDAV this is unfortunately impossible.

**Comment by tbrittoborges** [comment]
Thanks for the prompt answer.
Assuming we are admins/root, know `fileid` and the file owner, would it be possible to find the path to that file on the Nextcloud server file system?

**Reply by bigcat88** [reply, marked as answer]
```python3

    file_info = nc.files.by_id(123)
    print(file_info.full_path)
```
